server/src/main/java/org/elasticsearch/cluster/routing/allocation/ShardsAvailabilityHealthIndicatorService.java

```diff
@@ -252,9 +259,16 @@ public SimpleHealthIndicatorDetails getDetails() {
 
     }
 
-    private static String getTruncatedIndicesString(Set<String> indices) {
+    private static String getTruncatedIndicesString(Set<String> indices, Metadata clusterMetadata) {
         final int maxIndices = 10;
-        String truncatedIndicesString = indices.stream().limit(maxIndices).collect(joining(", "));
+        Comparator<String> comparePrioritiesAscendingComparator = Comparator.comparingInt(index -> clusterMetadata.index(index).priority());
+        // Reversing it because the higher the priority number, the higher the priority:
+        Comparator<String> comparePrioritiesDescendingComparator = comparePrioritiesAscendingComparator.reversed();
+        Comparator<String> comparePrioritiesThenNameComparaotr = comparePrioritiesDescendingComparator.thenComparing(index -> index);
```

Comment:
```suggestion
        Comparator<String> comparePrioritiesThenNameComparator = comparePrioritiesDescendingComparator.thenComparing(index -> index);
```

Reply:
Is `naturalOrder` better suited than using `thenComparing(index -> index)` ?

Reply:
Oh probably so. I originally had this lowercasing the names (which is why it's like this), but decided that was a bad idea.